stake-pool/program/src/processor.rs

```diff
@@ -941,70 +1047,241 @@ mod tests {
         // Create stake account
         let mut pool_info = create_stake_pool(fee);
 
-        let stake_account_key = Pubkey::new_unique();
-        let mut stake_account_account = Account::new(stake_balance, 100, &stake_program_id());
-        // TODO: Set stake account Withdrawer authority to pool_info.deposit_authority_key
+        let deposit_info = do_deposit(&mut pool_info, stake_balance);
 
-        // Create account to receive minted tokens
-        let (token_receiver_key, mut token_receiver_account) = create_token_account(
+        // Test stake pool balance
+        let state = State::deserialize(&pool_info.pool_account.data).unwrap();
+        match state {
+            State::Unallocated => panic!("Stake pool state is not initialized after deposit"),
+            State::Init(stake_pool) => {
+                assert_eq!(stake_pool.stake_total, stake_balance);
+                assert_eq!(stake_pool.pool_total, stake_balance);
+            }
+        }
+
+        // Test token balances
+        let user_token_state =
+            SplAccount::unpack_from_slice(&deposit_info.token_receiver_account.data)
+                .expect("User token account is not initialized after deposit");
+        assert_eq!(user_token_state.amount, user_token_balance);
+        let fee_token_state = SplAccount::unpack_from_slice(&pool_info.owner_fee_account.data)
+            .expect("Fee token account is not initialized after deposit");
+        assert_eq!(fee_token_state.amount, fee_token_balance);
+
+        // Test mint total issued tokens
+        let mint_state = SplMint::unpack_from_slice(&pool_info.mint_account.data)
+            .expect("Mint account is not initialized after deposit");
+        assert_eq!(mint_state.supply, stake_balance);
+    }
+    #[test]
+    fn test_withdraw() {
+        let mut pool_info = create_stake_pool_default();
+
+        let user_withdrawer_key = Pubkey::new_unique();
+
+        let stake_balance = sol_to_lamports(20.0);
+        let mut deposit_info = do_deposit(&mut pool_info, stake_balance);
+
+        let withdraw_amount = sol_to_lamports(5.0);
```

Comment:
nit: since `withdraw_amount` is in pool tokens and not lamports, could you hard-code a number instead?